crates/core/tedge_agent/src/agent.rs

```diff
@@ -267,17 +261,37 @@ impl SmAgent {
             );
         }
 
-        let mut mqtt_errors = mqtt.errors;
+        let request_journal = Journal::open(self.config.journal_path()).await?;
+        let persistance_store = request_journal.clone();
+
+        let mut mqtt = Connection::new(&self.config.mqtt_config).await?;
+        let mqtt_errors = mqtt.errors;
+        let mqtt_input = mqtt.received;
+        let mqtt_output = mqtt.published.clone();
+        let mut mqtt_responses = mqtt.published.clone();
+        let health_topic = self.config.response_topic_health.clone();
+        let errors_topics = self.config.errors_topic.clone();
+
         tokio::spawn(async move {
-            while let Some(error) = mqtt_errors.next().await {
-                error!("{}", error);
-            }
+            Self::process_mqtt_errors(mqtt_errors).await;
+        });
+        tokio::spawn(async move {
+            Self::process_subscribed_messages(
+                &request_journal,
+                mqtt_input,
+                mqtt_output,
+                health_topic,
+                errors_topics,
+            )
+            .await;
         });
 
-        let () = self.process_pending_operation(&mut mqtt.published).await?;
+        let () = self
```

Comment:
Shouldn't the pending operations be processed first, before processing any new messages? So, shouldn't the spawing of the `process_subscribed_messages` task happen after this step here?